### ISSUE TITLE
The order of the insert links in the editor to be presented in a logical order

### ISSUE THREAD

**Issue by clarkepaul** [silverstripe/silverstripe-admin#914]
As inserting links into the editor is one of the most used features of the CMS it makes sense that they are logically ordered so the most used link type is placed at the top. This issue is also covered in the Elemental link component design consistency issue but the order of the links presented makes more sense being sorted globally.

I suggest the order should be as per designs (maybe external link could be swapped with File link?). Note that "Anchor" links are to be integrated with internal/external links as part of other issues (see https://github.com/dnadesign/silverstripe-elemental/issues/589).

<img width="122" alt="Pasted_Image_29_07_19__12_43_PM" src="https://user-images.githubusercontent.com/555033/62015298-74267e00-b1fe-11e9-9b8d-cfe0d393bed1.png">

Design reference of link component for blocks but is similar https://projects.invisionapp.com/dsm/silver-stripe/silver-stripe/folder/components/5c68f96ffb26bb0018525d2b

# Pull Requests
* [x] silverstripe/silverstripe-asset-admin#975
* [x] silverstripe/silverstripe-admin#926
* [ ] https://github.com/silverstripe/silverstripe-cms/pull/2464

**Comment by maxime-rainville** [comment]
That's what's currently there.
![image](https://user-images.githubusercontent.com/1168676/62846755-9c49cd00-bd26-11e9-89dd-24f8256660ae.png)

Add link to "Anchor on another page" is not in the original mock up. I'll put it under external link for now. Let me know if you want a different order.

# Controlling the order of link option

The way those links are added is by using this bit of code.
```js
TinyMCEActionRegistrar.addAction(
  'sslink',
  {
    text: i18n._t('Admin.LINKLABEL_EMAIL', 'Link to email address'),
    onclick: (editorInst) => editorInst.execCommand(commandName),
    priority: 51,
  },
  editorIdentifier,
).addCommandWithUrlTest(commandName, /^mailto:/);
```

`priority` controls the order ... items with an higher priority gets displayed first.

I'll set the priority so there's room in-between each linking option in case a mad-scientist somewhere decides to add their own linking option and wants to put it at a specific place.

**Comment by clarkepaul** [comment]
The Anchor link was meant to be an extra field on the "Link to page" but never happened.

The thing that makes me the happiest is that "link to page" is at the top finally :) 
The order looks way better.

**Comment by maxime-rainville** [comment]
Got PRs for this. 

Here's what it looks like with the PRs.

![image](https://user-images.githubusercontent.com/1168676/62847725-e7b2aa00-bd2b-11e9-96e7-5996731949a2.png)

This also controls the order of the linking option in the context menu.
![image](https://user-images.githubusercontent.com/1168676/62847777-2ba5af00-bd2c-11e9-9552-09d07359a50f.png)

**Comment by robbieaverill** [comment]
Similar to what @ScopeyNZ reported the other day, I also am not seeing an updated order with the linked PRs checked out

**Comment by robbieaverill** [comment]
Ah, in Firefox it works. I was testing on Chrome (both on Mac).

**Comment by robbieaverill** [comment]
All merged into 4.5.x-dev
